docs/cdoc/qk-transpiler-layout.rst

```diff
@@ -0,0 +1,101 @@
+=================
+QkTranspileLayout
+=================
+
+.. code-block:: c
+
+   typedef struct QkTranspileLayout QkTranspileLayout
+
+The ``QkTranspileLayout`` type is used to module the permutations introduced by
+the transpiler. In general Qiskit's transpiler is unitary-preserving up to the
+initial layout and output permutations. The initial layout permutation is
+caused by setting and applying the initial layout (the mapping from virtual
+circuit qubits to physical qubits on the target) and the output permtuations
+are caused by swap gate insertion or permutation ellision prior to the initial
+layout being set in the transpiler pipeline. This type tracks these details and
+provide an interface to reason about these permutations.
```

Comment:
Please can we not refer to the layout as "a permutation" in new docs?  It's a relabelling - our insistence that we treat it as a permutation is what leads to the crazy "implicit ancilla" system.

Reply:
Sure I can change the wording here